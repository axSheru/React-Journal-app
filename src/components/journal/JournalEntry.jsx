import React from 'react';

export const JournalEntry = () => {
    return (
        <div className='journal__entry pointer'>
            <div
                className="journal___entry-picture"
                style={{
                    backgroundSize: 'cover',
                    backgroundImage: 'url(https://www.tonica.la/__export/1588444612791/sites/debate/img/2020/05/02/the_mandalorian_portada.jpg_1902800913.jpg)',
                }}
            >
            </div>
            <div className="journal__entry-body">
                <p className='journal__entry-title'>
                    Un nuevo dÃ­a.
                </p>
                <p className='journal__entry-content'>
                    Deserunt aliquip ut reprehenderit sint fugiat.
                </p>
            </div>
            <div className="journal__entry-date-box">
                <span>Tuesday</span>
                <h4>22</h4>
            </div>
        </div>
    );
}
