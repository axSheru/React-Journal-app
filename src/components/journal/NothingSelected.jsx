import React from 'react'

export const NothingSelected = () => {
    return (
        <div className='nothing__main-content'>
            <p>
                Select an entry
                <br />
                or create a new one!
            </p>
            <i className='far fa-star fa-4x mt-5'></i>
        </div>
    );
}
