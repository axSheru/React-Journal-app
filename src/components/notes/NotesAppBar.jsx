import React from 'react';

export const NotesAppBar = () => {
    return (
        <div className='notes__appbar'>
            <span>22 de marzo de 2022</span>
            <div>
                <button className='btn'>Picture</button>
                <button className='btn'>Save</button>
            </div>
        </div>
    );
}
