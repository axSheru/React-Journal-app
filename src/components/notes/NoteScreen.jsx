import React from 'react';
import { NotesAppBar } from './NotesAppBar';

export const NoteScreen = () => {
    return (
        <div className='notes__main-content'>
            <NotesAppBar />

            <div className="notes__content">
                <form className='notes__main-content'>
                    <input
                        type="text"
                        placeholder='Some awesome title'
                        className='notes__title-input'
                        autoComplete='off'
                    />
                    <textarea
                        placeholder='What happened today?'
                        className='notes__textarea'
                    ></textarea>
                    <div className="notes__image">
                        <img
                            src="https://i1.wp.com/www.lafosadelrancor.com/wp-content/uploads/2019/07/Sith-Trooper-El-Ascenso-de-Skywalker.jpg?fit=678%2C489&ssl=1"
                            alt="imagen"
                        />
                    </div>
                </form>
            </div>
        </div>
    );
}
