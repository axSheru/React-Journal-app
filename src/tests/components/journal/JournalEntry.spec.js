import configureStore from 'redux-mock-store';
import { mount } from "enzyme";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

const initState = {};

let store = mockStore( initState );

store.dispatch = jest.fn();

const nota = {
    id: 12,
    date: 0,
    title: 'This is',
    body: 'the way',
    url: 'https://algunlugar.com/fojo.jpg'
};

const wrapper = mount(
    <Provider store={ store } >
        <JournalEntry { ...nota } />
    </Provider>
);


describe('Pruebas en el componente JournalEntry.', () => {

    test('Debe de hacer match con el snapshot.', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de activar la nota.', () => {

        wrapper.find( '.journal__entry' ).prop( 'onClick' )();

        expect( store.dispatch ).toHaveBeenCalledWith(
            activeNote( nota.id, { ...nota } )
        );

    });

});