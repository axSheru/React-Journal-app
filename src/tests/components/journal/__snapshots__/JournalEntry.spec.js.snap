// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente JournalEntry. Debe de hacer match con el snapshot. 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal___entry-picture"
    style={
      Object {
        "backgroundImage": "url( https://algunlugar.com/fojo.jpg )",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      This is
    </p>
    <p
      className="journal__entry-content"
    >
      the way
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31st
    </h4>
  </div>
</div>
`;
